### PR TITLE
Merge individual Battery.[ch] files into Platform.[ch]

### PR DESCRIPTION
**Pull request by natoscott**
Consistent with everything else involving platform-specific
calls from core htop code.

### CODE REVIEW

File: BatteryMeter.c [hunk 1]
```diff
@@ -11,7 +11,7 @@ This meter written by Ian P. Hands (iphands@gmail.com, ihands@redhat.com).
 
 #include <math.h>
 
-#include "Battery.h"
+#include "Platform.h"
```
Comment (BenBE) [inline BatteryMeter.c:14]: Include sort order.

File: linux/Platform.c [hunk 2]
```diff
@@ -538,3 +541,323 @@ bool Platform_getNetworkIO(unsigned long int* bytesReceived,
...
+   closedir(dir);
+
+   *level = totalFull > 0 ? ((double) totalRemain * 100.0) / (double) totalFull : NAN;
+}
+
+static enum { BAT_PROC, BAT_SYS, BAT_ERR } Platform_Battery_method = BAT_PROC;
+
+static time_t Platform_Battery_cacheTime;
+static double Platform_Battery_cacheLevel = NAN;
+static ACPresence Platform_Battery_cacheIsOnAC;
```
Comment (BenBE) [inline linux/Platform.c:833]: Avoid mixing variables between functions. Better put those to the top.

File: openbsd/Battery.c [hunk 3]
```diff
@@ -1,74 +0,0 @@
-/*
-htop - openbsd/Battery.c
-(C) 2015 Hisham H. Muhammad
-(C) 2015 Michael McConville
```
Comment (BenBE) [inline openbsd/Battery.c:4]: Attribution lost after moving the code.

File: openbsd/Battery.h [hunk 4]
```diff
@@ -1,15 +0,0 @@
-#ifndef HEADER_Battery
-#define HEADER_Battery
-/*
-htop - openbsd/Battery.h
-(C) 2015 Hisham H. Muhammad
-(C) 2015 Michael McConville
```
Comment (BenBE) [inline openbsd/Battery.h:6]: Attribution lost after moving the code.

File: dragonflybsd/Battery.c [hunk 5]
```diff
@@ -1,30 +0,0 @@
-/*
-htop - dragonflybsd/Battery.c
-(C) 2015 Hisham H. Muhammad
-(C) 2017 Diederik de Groot
```
Comment (BenBE) [inline dragonflybsd/Battery.c:4]: Attribution lost after moving the code.

File: dragonflybsd/Battery.h [hunk 6]
```diff
@@ -1,15 +0,0 @@
-#ifndef HEADER_Battery
-#define HEADER_Battery
-/*
-htop - dragonflybsd/Battery.h
-(C) 2015 Hisham H. Muhammad
-(C) 2017 Diederik de Groot
```
Comment (BenBE) [inline dragonflybsd/Battery.h:6]: Attribution lost after moving the code.

**Review by BenBE — CHANGES_REQUESTED** [review]

File: linux/Platform.c (hunk 2 above)
Comment (natoscott) [reply linux/Platform.c:833]: Will do - thanks @BenBE

File: openbsd/Battery.c (hunk 3 above)
Comment (natoscott) [reply openbsd/Battery.c:4]: openbsd/Platform.c already has these authors copyright annotations listed.  Likewise for the other cases you mentioned.

**Review by BenBE — APPROVED** [review]

**Comment by fasterit** [comment]
Merged with 0eb3c75

**Pull request closed without merging**